src/Twitter.php

```diff
@@ -0,0 +1,335 @@
+            'token' => $config->getAccessToken(),
+            'token_secret' => $config->getAccessTokenSecret(),
+        ]);
+        $stack->push($middleware);
+
+        return new HttpClient([
+            'handler' => $stack,
+            'auth' => 'oauth',
+        ]);
+    }
+
+    /**
+     * @param array  $params
+     * @param string $requestMethod
+     *
+     * @return array
+     */
+    private function getRequestOptions(array $params, string $requestMethod): array
+    {
+        unset($params[self::KEY_FORMAT]);
+
+        $paramsKey = $requestMethod === self::REQUEST_METHOD_POST ? self::KEY_FORM_PARAMS : self::KEY_QUERY;
+
+        return [
+            $paramsKey => $params,
+        ];
+    }
+
+    /**
+     * @param string $host
+     * @param string $version
+     * @param string $name
+     * @param string $extension
+     *
+     * @return string
+     */
+    private function buildUrl(string $host, string $version, string $name, string $extension): string
+    {
+        return sprintf(self::URL_FORMAT, $host, $version, $name, $extension);
+    }
+
+    /**
+     * @param GuzzleException $exception
+     *
+     * @return TwitterRequestException
+     */
+    private function transformClientException(GuzzleException $exception): TwitterRequestException
+    {
+        /** @var null|Response $response */
+        $response = method_exists($exception, 'getResponse') ? $exception->getResponse() : null;
+        $responseCode = !empty($response) ? $response->getStatusCode() : null;
+
+        switch ($responseCode) {
+            case self::RESPONSE_CODE_400:
+                return BadRequestException::fromClientResponse($response, $exception);
+            case self::RESPONSE_CODE_401:
+                return UnauthorizedRequestException::fromClientResponse($response, $exception);
+            case self::RESPONSE_CODE_403:
+                return ForbiddenRequestException::fromClientResponse($response, $exception);
+            case self::RESPONSE_CODE_404:
+                return NotFoundException::fromClientResponse($response, $exception);
+            case self::RESPONSE_CODE_420:
+                return RateLimitedException::fromClientResponse($response, $exception);
+            default:
+                return TwitterRequestException::fromClientResponse($response, $exception);
+        }
+    }
+
+    /**
+     * @param Response $response
+     * @param string   $format
+     *
+     * @return mixed|string
+     */
+    private function formatResponse(Response $response, string $format)
+    {
+        $body = (string)$response->getBody();
+
+        switch ($format) {
+            case self::RESPONSE_FORMAT_JSON:
+                return $body;
+            case self::RESPONSE_FORMAT_ARRAY:
+                return json_decode($body, true);
+            case self::RESPONSE_FORMAT_OBJECT:
+            default:
+                return json_decode($body, false);
```

Comment:
Do we want to keep the default as object? TBH would prefer it as array unless we are going to create DTOs for the responses.

Reply:
Either is fine actually. My only argument for `object` is that it was this by default before.

Reply:
Fair point. Happy to leave it as in for the moment.